config/locales/en.yml

```
@@ -301,8 +301,8 @@ en:
     tenancy:
       length:
         fixed_term_not_required: "You must only answer the length of the tenancy if it's fixed-term"
-        shorthold: "Enter a tenancy length between 2 and 99 years for a Fixed Term – Assured Shorthold Tenancy (AST)"
-        secure: "Enter a tenancy length between 2 and 99 years (or don't specify the length) for a Secure (including flexible) tenancy"
+        shorthold: "Enter a tenancy length between %{min_tenancy_length} and 99 years for a %{needs_type} %{rent_type} Fixed Term – Assured Shorthold Tenancy (AST)"
+        secure: "Enter a tenancy length between %{min_tenancy_length} and 99 years (or don't specify the length) for a %{needs_type} %{rent_type} Secure (including flexible) tenancy"
```

Do we actually need to include the `%{needs_type} %{rent_type}` part of this, here?

---

```suggestion
        secure: "Enter a tenancy length between %{min_tenancy_length} and 99 years (or don't specify the length) for a Secure (including flexible) tenancy of this type"
```

... perhaps we could make it generic like this?

---

If it doesn't include it it's not technically accurate, because you can have an AST with a tenancy length of 1 year, but only when it is General Needs and Affordable or Social Rent

---

I think my worry is that a user will complete one log with a tenancy length of 1 year that is accepted, then another that hits this error, and they will be confused why the error message looks like it should apply to the first log too.
The tenancy type, rent type, and needs type all affect whether this error appears so I think the user should be told about them if possible.

---

Yes, that's a fair challenge.